### ISSUE TITLE
Enable decorator support in options

### ISSUE THREAD

**Issue by ashwinr** [palantir/eclipse-typescript#255]
The following error is shown by the plugin when decorators are used in 1.5:

`Experimental support for decorators is a feature that is subject to change in a future release. Specify '--experimentalDecorators' to remove this warning.`

Turning on decorator support should be an option in the compiler settings that will in turn enable this flag.

**Comment by derekcicerone** [comment]
Already supported in the develop release.

**Comment by faisalnet5** [comment]
@derekcicerone: I know this topic is already closed but unfortunately today i had the same problem:
<code>
Experimental support for decorators is a feature that is subject to change in a future release. Specify '--experimentalDecorators' to remove this warning.
</code>
I am using latest 'DefinitelyTyped/angular2/angular2.d.ts' with ES5 (module code: commonjs). I tried ES6 where that above error is gone but having project build error: 'Path must include project and resource name'

**Comment by derekcicerone** [comment]
Did you try enabling the option to support decorators in the project settings? You'll also need to be on version 1.5 or later.

Derek
Sent from my iPhone

On Aug 12, 2015, at 4:45 PM, faisalnet5 <notifications@github.com<mailto:notifications@github.com>> wrote:


@derekcicerone<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_derekcicerone&d=BQMCaQ&c=izlc9mHr637UR4lpLEZLFFS3Vn2UXBrZ4tFb6oOnmz8&r=2K9mkcA15muZoiO7gygKNOjy-FVfnllq4-gUDYqQcTM&m=pS1DMsUvV-1eR_TfYyHYsa33UZu_8KFyTA-f7XdlksQ&s=4QjGOq4YFHyUTjqaftXI1xlh7k58ThG2Lp96pkk6mwY&e=>: I know this topic is already closed but unfortunately today i had the same problem:

Experimental support for decorators is a feature that is subject to change in a future release. Specify '--experimentalDecorators' to remove this warning.

I am using latest 'DefinitelyTyped/angular2/angular2.d.ts' with ES5 (module code: commonjs). I tried ES6 where that above error is gone but having project build error: 'Path must include project and resource name'

**Comment by faisalnet5** [comment]
But how? :)

Well, I am using Eclipse. I did install TypEcs (TypeScript IDE for Eclipse) and 'eclipse-typescript' plugin.
On .ts file save now it is making two file (a) .js and (b) .js.map. No making the file name 'tsconfig.json'. 

Currently my project structure is as follows:

"WebContent"
   * ----js
   * ----lib
     * ----angular2
       * ----angular2.d.ts
     * ----es6-promise
        * ----es6-promise.d.ts
     * ----rx
        * ----rx.d.ts
        * ----rx-lite.d.ts
     * ----app.js
     * ----app.js.map
     * ----app.ts
   * ----META-INF
   * ----WEB-INF
   * ----index.html

```javascript
# index.html contents:
-------------------------------------------
<!DOCTYPE html>
<html>
<head>
  <title>Angular 2 Quickstart</title>
  <script src="https://github.jspm.io/jmcriffey/bower-traceur-runtime@0.0.87/traceur-runtime.js"></script>
  <script src="https://jspm.io/system@0.16.js"></script>
  <script src="https://code.angularjs.org/2.0.0-alpha.28/angular2.dev.js"></script>
</head>
<body>
	<my-app></my-app>
	<script>System.import('js/app');</script>
</body>
</html>

# app.ts contents:
------------------------------------------------------------
/// <reference path="../js/lib/angular2/angular2.d.ts" />
import {Component, View, bootstrap} from 'angular2/angular2';
// Annotation section
@Component({
    selector: 'my-app'
})
@View({
   template: '<h1>Hello {{ name }}</h1>'
})
// Component controller
class MyAppComponent {                      //Here the Error comes
    name: string;
    constructor() {
        this.name = 'Faisal 123';
    }
}
bootstrap(MyAppComponent);
```
Finally, although the error is there but still I was able to deploy this sample project in Tomcat and populate the page with right result.

**Comment by derekcicerone** [comment]
You'll need to right-click on the project in Eclipse and change the Eclipse settings. Look for "Enable experimental support for ES7 decorators."

**Comment by faisalnet5** [comment]
Nope...I am not seeing "Enable experimental support for ES7 decorators." anywhere in my Eclipse :ghost: Btw, here using Eclipse Luna.

Well under 'Configure' right now I kept the config as follows:
(a) Enable TypeScript Builder
(b) Remove TypeScript Support
(Note: These are the only two options I have related to TypeScript)

Now, if I 'Enable TypeScript Builder' then immediately it is throwing more that 1K 'Duplication' errors in a blink of eyes ;)

**Comment by derekcicerone** [comment]
The settings screen you are looking for is like this:
<img width="633" alt="screen shot 2015-08-13 at 3 27 24 pm" src="https://cloud.githubusercontent.com/assets/1081904/9259538/d7a736ca-41cf-11e5-8978-71ad7f015cff.png">

**Comment by faisalnet5** [comment]
Sorry @derekcicerone :( No luck....I tried all those options before.....may be Eclipse hates me because see the following picture....ExampleFirst is showing the Error but ExampleSecond is not! How come? Believe me....they r twins. I guess, I need a :coffee: ... Btw, I also did changes to all my previous setting exactly like the way ur shared image is showing.....but still same result :disappointed:  

![screen](https://cloud.githubusercontent.com/assets/9747030/9259893/6ff62de4-41d2-11e5-839f-9aebe4fc4374.PNG)

**Comment by derekcicerone** [comment]
I think you are running multiple TypeScript plug-ins. You would need to uninstall TypEcs.
